contracts/extensions/BasicIssuanceExtension.sol

```diff
@@ -0,0 +1,248 @@
+
+        address methodologist = delegatedManager.methodologist();
+        address ownerFeeRecipient = delegatedManager.ownerFeeRecipient();
+
+        uint256 ownerTake = totalFees.preciseMul(delegatedManager.ownerFeeSplit());
+        uint256 methodologistTake = totalFees.sub(ownerTake);
+
+        if (ownerTake > 0) {
+            delegatedManager.transferTokens(address(_setToken), ownerFeeRecipient, ownerTake);
+        }
+
+        if (methodologistTake > 0) {
+            delegatedManager.transferTokens(address(_setToken), methodologist, methodologistTake);
+        }
+
+        emit FeesDistributed(address(_setToken), ownerFeeRecipient, methodologist, ownerTake, methodologistTake);
+    }
+
+    /**
+     * ONLY OWNER: Initializes BasicIssuanceExtension to the DelegatedManager.
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeExtension(IDelegatedManager _delegatedManager) external {
+        ISetToken setToken = _delegatedManager.setToken();
+
+        require(
+            managerCore.isFactory(msg.sender) || 
+            address(_delegatedManager) == setToken.manager(),
+            "Must be factory or input must be SetToken manager"
+        );
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        setManagers[setToken] = _delegatedManager;
+
+        _delegatedManager.initializeExtension();
+
+        ExtensionInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY OWNER: Initializes BasicIssuanceExtension to the DelegatedManager and BasicIssuanceModule to the SetToken
+     *
+     * @param _delegatedManager             Instance of the DelegatedManager to initialize
+     * @param _maxManagerFee                Maximum fee that can be charged on issue and redeem
+     * @param _managerIssueFee              Fee to charge on issuance
+     * @param _managerRedeemFee             Fee to charge on redemption
+     * @param _feeRecipient                 Address to send fees to
+     * @param _managerIssuanceHook          Instance of the Manager Contract with the Pre-Issuance Hook function
+     */
+    function initializeModuleAndExtension(
+        IDelegatedManager _delegatedManager,
+        uint256 _maxManagerFee,
+        uint256 _managerIssueFee,
+        uint256 _managerRedeemFee,
+        address _feeRecipient,
+        address _managerIssuanceHook
+    ) 
+        external 
+    {
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        ISetToken setToken = _delegatedManager.setToken();
+
+        require(setToken.isPendingModule(address(issuanceModule)), "BasicIssuanceModule must be pending");
```

Comment:
I think this check might be duplicated, e.g it gets called as part of DebtIssuance.initialize here:

+ [DebtIssuance.initialize L315][1]
+ [BaseModule: modifier logic L233][2]

[1]: https://github.com/SetProtocol/set-protocol-v2/blob/87e48a7e0e969340e74c068f701ef8738827afd1/contracts/protocol/modules/DebtIssuanceModule.sol#L315

[2]: https://github.com/SetProtocol/set-protocol-v2/blob/87e48a7e0e969340e74c068f701ef8738827afd1/contracts/protocol/lib/ModuleBase.sol#L233-L236